include/asterisk/app.h

```diff
@@ -1196,9 +1196,54 @@ struct ast_group_info;
 /*! \brief Split a group string into group and category, returning a default category if none is provided. */
 int ast_app_group_split_group(const char *data, char *group, int group_max, char *category, int category_max);
 
+/* Create group meta if it doesn't exist  */
+int ast_app_group_add_meta_ifneeded(const char *group, const char *category);
+
+/* This removes the the group metadata for a group@category if there are no more channels in the group@category */
+int ast_app_group_remove_meta_ifneeded(const char *group, const char *category);
+
+/*! \brief Remove channel assignments for the specified group  */
+int ast_app_group_remove_all_channels(const char *group, const char *category);
+
+/*! \brief Rename a group@category while retaining all the channel memberships
+ * \retval -1 on Error
+ * \retval 0 On zero rename operations performed (nothing found to rename)
+ * \retval >=1 On 1 or more rename operations performed
```

Comment:
This one might be clearer as `\return Number of rename operations performed`. It would also cover the 0 case

Reply:
Fixed